### PR TITLE
📦 Webpack 5 compatibility + Attempting to fix polyfill issues

### PR DESCRIPTION
**Pull request by MikaelCarpenter**
## Links
- Issue #95 gives a lot of the background context
- Webpack `exports` field [documentation](https://webpack.js.org/guides/package-exports/#root)

## Why
Allow people to utilize this module in their projects if they're building with Webpack 5 and targeting certain browser compatibilities that require additional polyfills

## What
- Update the `"exports"` field in `package.json` to include an export for `fragment-generation-utils.js`
- Provide a more robust `rollup` config that includes `async`/`await` polyfills

## Impact
The second commit changes the size and build time:
- generating the `dist` of `text-fragments.js` jumps from 300ms to 3s on my machine
- the resulting size of `dist/text-fragments.js` jumps from 9kb to 47kb

As I mentioned in the issue, the second fix is much more cumbersome, so if you'd prefer another approach, just let me know 👍

### CODE REVIEW

**Comment by MikaelCarpenter** [comment]
@googlebot I signed it!

**Comment by tfmar** [comment]
Hi Mikael, thanks for your clear and detailed report and pull request.

# Exports:
I think this makes sense, but I'm going to ask Thomas (tomayac) to review your patch, because he depends more on the exports than I do.

# Rollup / Regenerate:
I'm not super familiar with rollup; do you know what exactly is causing `regenerate` to get included in your build? (I mean, obviously it has to do with the usage of `async`/`await`/ `yield`, but what step in the compilation is deciding that particular polyfill is needed and adding it?) I'm also curious if you happen to know what you're targeting (ECMAScript version or browser version) that necessitates polyfilling these features.

I definitely think we should try and avoid including polyfills if possible; the jump from 9kb to 47kb is fairly significant for us.

The good news is, our usage of `async` and `yield` is super basic and minimal. In your case, since you're only interested in `fragment-generation-utils` and not `text-fragments`, I think there's just one usage of `yield` that's going into your build. (It's in `text-fragment-utils.js`.) IIRC we're immediately flattening the elements yielded by that generator anyway, so it's quite trivial. We may well be able to just replace this usage and get your build working. If that works out we might not even need the new `rollup.config.js` at all.

**Comment by MikaelCarpenter** [comment]
> I'm going to ask Thomas (tomayac) to review your patch

Awesome 👍 

> what step in the compilation is deciding that particular polyfill is needed and adding it? I'm also curious if you happen to know what you're targeting

Great question, I assume it's `babel-loader` in my webpack config:
```javascript
// Webpack config
      {
        test: /\.(js|jsx)$/,
        use: [
          {
            loader: 'source-map-loader',
          },
          {
            loader: 'babel-loader',
            options: {
              rootMode: 'upward',
            },
          },
        ],
        exclude: /node_modules/,
      },

// Separate babel config has:
    [
      "@babel/preset-env",
      {
        "targets": {
          "browsers": [
            ">0.25%",
            "not ie 11",
            "not op_mini all",
            "not dead"
          ]
        },
        useBuiltIns: 'usage',
        corejs: 3,
      },
    ],
```

Which now that I think about it, I can update that `exclude` and try to transpile this module on my end if necessary. 

> I definitely think we should try and avoid including polyfills if possible; the jump from 9kb to 47kb is fairly significant for us.

Yup, totally makes sense 👍 I'll go ahead and pull that commit out for now and then attempt the transpile on my end to get it working for myself.

File: package.json [hunk 1]
```diff
@@ -5,7 +5,10 @@
   "main": "./dist/text-fragments.js",
   "browser": "./dist/text-fragments.js",
   "module": "./dist/text-fragments.js",
-  "exports": "./dist/text-fragments.js",
+  "exports": {
+    ".": "./dist/text-fragments.js",
+    "./dist/fragment-generation-utils.js": "./dist/fragment-generation-utils.js"
```
Comment (tomayac) [inline package.json:10]:
```suggestion
    "./dist/fragment-generation-utils.js": "./dist/fragment-generation-utils.js",
    "./package.json": "./package.json"
```

**Review by tomayac — APPROVED** [review]
LGTM, approval

**Pull request merged**